### ISSUE TITLE
 Request failed - A specified logon session does not exist. It may already have been terminated System.ComponentModel.Win32Exception (0x80004005): A specified logon session does not exist. It may already have been terminated

### ISSUE THREAD

**Issue by ATechGuyME** [webprofusion/certify#294]
I'm running Certify 3.0.11.30336 on server 2012 R2

I read through a million articles on getting past the .well-known and extensionless file handling. I ended up being able to create a certificate by using acme-win but I wanted to see if after that, I could use certify since it could be a lot simpler if it worked OOTB.

I have modified my default site web.config file to contain the mimemap for "." to be text/plain, so when certify runs, it was throwing an IIS error were the mimemap had already been configured, as it was putting a new web.config file in the acme-challenge folder and it was conflicting with the root definition. Noting this, I removed the entry from the root and the error was cleared but now I get this new error. I CAN read my configcheck file from an external address, so I know the URL for mydomain.com/.well-known/acme-challenge/configcheck is ok. the new error entry shows:

2018-04-24 15:20:08.942 -04:00 [INF] Default Web Site: Request failed - A specified logon session does not exist. It may already have been terminated System.ComponentModel.Win32Exception (0x80004005): A specified logon session does not exist. It may already have been terminated
   at Microsoft.Web.Management.Utility.HttpApiWrapper.CreateSSLBinding(IPEndPoint endPoint, String hostName, HTTP_SERVICE_CONFIG_SSL_PARAM_MANAGED allSSLData, SslFlags sslFlags)
   at Microsoft.Web.Management.Utility.HttpApiWrapper.CreateSSLBinding(IPEndPoint endPoint, String hostName, Byte[] certificateHash, String certificateStoreName, SslFlags sslFlags)
   at Microsoft.Web.Administration.BindingManager.BindingTransaction.Commit()
   at Microsoft.Web.Administration.BindingManager.Save()
   at Microsoft.Web.Administration.ServerManager.CommitChanges()
   at Certify.Management.IISManager.InstallCertificateforBinding(String certStoreName, Byte[] certificateHash, Site site, String host, Int32 sslPort, Boolean useSNI, String ipAddress, Boolean alwaysRecreateBindings) in D:\Work\GIT\certify\src\Certify.Core\Management\IISManager.cs:line 644
   at Certify.Management.IISManager.<InstallCertForRequest>d__31.MoveNext() in D:\Work\GIT\certify\src\Certify.Core\Management\IISManager.cs:line 484
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Certify.Management.CertifyManager.<PerformCertificateRequestProcessing>d__52.MoveNext() in D:\Work\GIT\certify\src\Certify.Core\Management\CertifyManager.cs:line 717
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Certify.Management.CertifyManager.<PerformCertificateRequest>d__51.MoveNext() in D:\Work\GIT\certify\src\Certify.Core\Management\CertifyManager.cs:line 420

It DOES appear to have fetched a valid certificate (and added "(certify)" to the name, but perhaps this error is just a problem with the Binding?

I have since cleared out all but the certify certificate, and I would be happy to try renewing the current certificate, but it's still valid, so certify doesn't want to renew it.

If there is anything else I can do  or provide, please let me know.

**Comment by webprofusion-chrisc** [comment]
Thanks for your perseverance!

- The first issue around config is likely to be an issue with our current default web.config, and I suspect 3.x has a bug related to how we auto-configure. 4.x is in the works and has a different system for auto config, with better defaults based on the most likely versions of IIS. This version is available in alpha test and a full release is coming soon.
- For the second issue, this can either be a transient issue requiring a server restart (not directly related to Certify) or this is a permission issue related to IIS trying to access the private key for the certificate. Our default configuration has the background service running as local system, have you modified that in any way or do you have any restriction in place for the local system account? You can test out the binding step again by choosing 'Re-apply certificate to bindings' under 'Advanced' - this avoids re-requesting the certificate if you already have one.

**Comment by webprofusion-chrisc** [comment]
Did you find out if a reboot fixed it?

**Comment by ATechGuyME** [comment]
Yes! I'm happy to report that doing a manual request to obtain a cert now, after the reboot, it works flawlessly. Thanks for the follow up. 

I'm not sure how much help I could be with an Alpha release, but once you're ready for "early" beta testers, please add my name to your list. Most of my clients are on 2008 R2 (typically SBS 2011 variants) or later.

**Comment by ATechGuyME** [comment]
FWIW, I just tried it on a client system running SBS 2011 Essentials and the only thing I've found I needed to do is to set the Local Hosts. file to include an entry for the external CNAME that we use, so that the firewall wouldn't have issues. Many of my clients use DYN.com for a dyndns.biz domain, and then we map a CNAME in the client's regular domain to point to the dyndnz.biz A record. 

Certify worked out of the box on my first client. Sweet.
